app/Repositories/Activity/ActivityRepository.php

```diff
@@ -157,6 +157,14 @@ public function advanceSearchForm($data)
             $searchResultQuery = $searchResultQuery->postFilter('term', ['_index' => $data['model']]);
         }
 
+        if (isset($data['isPublic']) && is_bool($data['isPublic'])) {
```

Comment:
Same here.

Reply:
Removed check for "is_bool".

Reply:
Hmm... I can still see is_bool check here.

Reply:
"is_bool" is needed here,
I cannot depend on person call this function to ensure that value is indeed boolean.